src/OpenTelemetry.AutoInstrumentation/Loading/LazyInstrumentationBuilders.cs

```diff
@@ -0,0 +1,60 @@
+// <copyright file="LazyInstrumentationBuilders.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using OpenTelemetry.AutoInstrumentation.Configuration;
+
+namespace OpenTelemetry.AutoInstrumentation.Loading
+{
+    internal class LazyInstrumentationBuilders
+    {
+        // Add here all of the available detectors
+        private static readonly IReadOnlyDictionary<TracerInstrumentation, Func<AssemblyLoadDetector>> _defaultDetectors
+            = new Dictionary<TracerInstrumentation, Func<AssemblyLoadDetector>>()
+        {
+            { TracerInstrumentation.AspNet, () => new AspNetCoreDetector() }
```

Comment:
If we want this to work with ASP.NET on .NET Framework, I think we may also need to add a mapping to a potential `new AspNetDetector()`

Reply:
Thanks, resolving